packages/components/src/components/common/frameProps.tsx

```diff
@@ -11,12 +11,24 @@ type Margin = {
 
 export type FrameProps = {
     margin?: Margin;
-    maxWidth?: string;
+    width?: string | number;
+    maxWidth?: string | number;
+    height?: string | number;
+    maxHeight?: string | number;
 };
 
 export type TransientFrameProps = TransientProps<FrameProps>;
 
-export const withFrameProps = ({ $margin, $maxWidth }: TransientFrameProps) => {
+const getValueWithUnit = (value: string | number) =>
+    typeof value === 'string' ? value : `${value}px`;
```

Comment:
We want to allow also string for values like `100%`

Reply:
If you want to do some TS magic I think it is possible to type and percentage value by using [template-string typing](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html):

`type CssValue = `${number}%` | number`

but I havent tested it